frontend/app/services/benefit-application-service.server.ts

```diff
@@ -103,24 +104,10 @@ function createBenefitApplicationService() {
 
     instrumentationService.countHttpStatus('http.client.application-history-api.applications.gets', response.status);
 
-    const applicationsSchema = z.array(
-      z.object({
-        AppicationId: z.string().optional(),
-        SubmittedDate: z.string().optional(),
-        ApplicationStatus: z.string().optional(),
-        ConfirmationCode: z.string().optional(),
-      }),
-    );
-
     const data = await response.json();
     log.trace('Applications for user id [%s]: [%j]', userId, data);
-    const applications = applicationsSchema.parse(data).map((application) => ({
-      id: application.AppicationId,
-      submittedOn: application.SubmittedDate,
-      status: application.ApplicationStatus,
-      confirmationCode: application.ConfirmationCode,
-    }));
 
+    const applications = toBenefitApplication(applicationListSchema.parse(data));
```

Comment:
I think it might be worth adding a `TODO` somewhere in this function to indicate that the mapping is based on the assumption that the API will return a response similar in structure to the Interop's `/benefit-application` request and that the actual API endpoint isn't available yet.

Reply:
Fixed:  a3363de42a431757737fa4c338f8126b2aca30e5